BGAnimations/ScreenGameplay overlay/TrackExScoreJudgments.lua

```diff
@@ -0,0 +1,118 @@
+}
+
+local valid_tns = {
+    W1 = true,
+    W2 = true,
+    W3 = true,
+    W4 = true,
+    W5 = true,
+    Miss = true,
+    HitMine = true
+}
+
+local valid_hns = {
+    LetGo = true,
+    Held = true
+}
+
+return Def.Actor{
+	JudgmentMessageCommand=function(self, params)
+		if params.Player ~= player then return end
+        local count_updated = false
+		if params.HoldNoteScore then
+            local HNS = ToEnumShortString(params.HoldNoteScore)
+            -- Only track the HoldNoteScores we care about
+            if valid_hns[HNS] then
+                if not stats:GetFailed() then
+                    ex_counts[HNS] = ex_counts[HNS] + 1
+                    count_updated = true
+                end
+            end
+        end
+
+		if params.TapNoteScore then
+            local TNS = ToEnumShortString(params.TapNoteScore)
+
+            if SL.Global.GameMode == "ITG" then
+                if TNS == "W1" then
+                    -- Check if this W1 is actually in the W0 window
+                    local is_W0 = IsW0Judgment(params, player)
+                    if is_W0 then
+                        if not stats:GetFailed() then
+                            ex_counts.W0 = ex_counts.W0 + 1
+                            count_updated = true
+                        end
+                        ex_counts.W0_total = ex_counts.W0_total + 1
+                    else
+                        if not stats:GetFailed() then
+                            ex_counts.W1 = ex_counts.W1 + 1
+                            count_updated = true
+                        end
+                    end
+                else
+                    -- Only track the TapNoteScores we care about
+                    if valid_tns[TNS] then
+                        if not stats:GetFailed() then
+                            ex_counts[TNS] = ex_counts[TNS] + 1
+                            count_updated = true
+                        end
+                    end
+                end
+            else
+                adjusted_TNS = TNS
+                -- In FA+ mode, we need to shift the windows up 1 so that the key we're using is accurate.
+                -- E.g. W1 window becomes W0, W2 becomes W1, etc.
+                if TNS ~= "Miss" then
+                    adjusted_TNS = "W"..(tonumber(TNS:sub(-1))-1)
+                end
+
+                if not stats:GetFailed() then
+                    ex_counts[adjusted_TNS] = ex_counts[adjusted_TNS] + 1
+                    count_updated = true
+                end
+            end
+
+            if count_updated then
+                -- Broadcast so other elements on ScreenGameplay can process the updated count.
+                MESSAGEMAN:Broadcast("ExCountsChanged", { Player=player, ExScore=CalculateExScore(player, ex_counts) })
+            end
+		end
+	end,
+	OffCommand=function(self)
+		local storage = SL[ToEnumShortString(player)].Stages.Stats[SL.Global.Stages.PlayedThisGame + 1]
+		storage.ex_counts = ex_counts
```

Comment:
Can we do this in the `OnCommand` instead? That way we wouldn't have to pass the counts into `CalculateExScore()`. because the counts are always available in the same location.

Reply:
Done.